### PR TITLE
GitHub : Build/publish Docker image for releases.

### PR DESCRIPTION
**Pull request by themissingcow**
... as GitHub Actions  are 🐙 👍

### CODE REVIEW

File: .github/workflows/dockerimage.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        IMAGE_ID=docker.pkg.github.com/${{ github.repository }}/$IMAGE_NAME
+        
+        # Strip git ref prefix from version
+        VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+        
+        # Strip "v" prefix from tag name
+        [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+        
+        # Use Docker `latest` tag convention
+        [ "$VERSION" == "master" ] && VERSION=latest
```
Comment (johnhaddon) [inline .github/workflows/dockerimage.yml]: Remind me, why would we do this? And if we upload the next image as `latest`, how do we refer to it as `1.1.0` afterwards?

**Review by johnhaddon — COMMENTED** [review]
What's the plan for `build-docker.py` now? Should we make `--upload` upload the package to GitHub, or should we remove `--upload`?

File: .github/workflows/dockerimage.yml (hunk 1 above)
Comment (themissingcow) [reply .github/workflows/dockerimage.yml]: I borrowed this template from GitHub Actions, and left some of the 'compatibility' code in, incase we wanted to do non-release CI ever. This just calls it 'latest' if it's a build from `master` (as opposed to a release) - that may have been triggered by a merge/nightly.

We could strip it (and some other compatibility  code, eg: `v` stripping) that we're not using right now.

**Comment by themissingcow** [comment]
> What's the plan for `build-docker.py` now? Should we make `--upload` upload the package to GitHub, or should we remove `--upload`?

I was going to leave it for a mo, until GitHub has proved itself. I think we'd maybe keep `--upload` and point it to GitHub, incase CI fails ever for 'other reasons' and we ever need to publish a build manually. We still need the script for local testing.

File: .github/workflows/dockerimage.yml (hunk 1 above)
Comment (johnhaddon) [reply .github/workflows/dockerimage.yml]: Yes, please strip. It's misleading to have code in here which we don't expect to do anything.

Comment (themissingcow) [reply .github/workflows/dockerimage.yml]: Removed in https://github.com/GafferHQ/build/pull/16/commits/66b73e8b3b5683164965643e61a4604745637e99

**Pull request merged**